### PR TITLE
Remove a wrong multiplier on relocation offset computation

### LINKED ISSUE #56800: type validation failed: encountered NULL reference at [2].0
Minimized as best I could:
```rust
fn main() {
    let mut headers = Vec::new();
    headers.extend_from_slice(&[("", ""); 3]);
}
```
With `rustc 1.32.0-nightly (f4a421ee3 2018-12-13)`
Full error:
```
ERROR 2018-12-14T02:46:10Z: rustc_mir::interpret::validity: Bits { size: 8, bits: 0 } is not aligned to Align { pow2: 0 }
error[E0080]: it is undefined behavior to use this value
 --> src/main.rs:1:1
  |
1 | / fn main() {
2 | |     let mut headers = Vec::new();
3 | |     headers.extend_from_slice(&[("", ""); 3]);
4 | | }
  | |_^ type validation failed: encountered NULL reference at [2].0
  |
  = note: The rules on what exactly is undefined behavior aren't clear, so this check might be overzealous. Please open an issue on the rust compiler repository if you believe it should not be considered undefined behavior
```

### PR DESCRIPTION
**Pull request by oli-obk**
r? @RalfJung 

fixes #56800

### CODE REVIEW

File: src/librustc_mir/interpret/memory.rs [hunk 1]
```diff
@@ -707,10 +707,10 @@ impl<'a, 'mir, 'tcx, M: Machine<'a, 'mir, 'tcx>> Memory<'a, 'mir, 'tcx, M> {
                 new_relocations.extend(
                     relocations
                     .iter()
-                    .map(|&(offset, reloc)| {
-                    (offset + dest.offset - src.offset + (i * size * relocations.len() as u64),
-                     reloc)
-                    })
+                    .map(|&(offset, reloc)| (
+                        offset + dest.offset - src.offset + (i * size),
```
Comment (RalfJung) [inline src/librustc_mir/interpret/memory.rs]: Might be clearer to do something like `let dest_offset = dest.offset + (i * size);` and then `offset + dest_offset - src.offset` to show that the `i * size` is part of the destination offset (I started at this for a bit wondering where it is coming from).

**Comment by RalfJung** [comment]
Good catch!

r=me with the nit fixed.

**Comment by pnkfelix** [comment]
discussed at T-compiler meeting. beta-accepting.

**Pull request merged**